### PR TITLE
Replace `GatewayIntent#GUILD_EMOJIS_AND_STICKERS` with `GUILD_EXPRESSIONS`

### PR DESCRIPTION
**Pull request by freya022**
[contributing]: https://jda.wiki/contributing/contributing/

## Pull Request Etiquette

- [X] I have checked the PRs for upcoming features/bug fixes.
- [X] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [X] Library interface (affecting end-user code) 
- [X] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

Updates the intent to match Discord's name, will be clearer for #2753

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/requests/GatewayIntent.java [hunk 1]
```diff
@@ -100,8 +101,16 @@ public enum GatewayIntent
     GUILD_MODERATION(2),
     /**
      * Custom emoji and sticker add/update/delete events.
+     *
+     * @deprecated Replaced with {@link #GUILD_EXPRESSIONS}
      */
+    @ForRemoval(deadline = "5.2.0")
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/requests/GatewayIntent.java:107]:
```suggestion
    @ForRemoval(deadline = "5.3.0")
    @ReplaceWith("GUILD_EXPRESSIONS")
```

**Comment by MinnDevelopment** [comment]
> If the user does not migrate to the new intent, and has activated cache flags that require the GUILD_EXPRESSIONS intents, the cache flags will get disabled and a warning will be logged.

I don't think it will be disabled, since the check will include both regardless. I'll consider writing a test for this.

**Review by MinnDevelopment — CHANGES_REQUESTED** [review]

**Comment by freya022** [comment]
True, as it uses the raw intents, it sees both the old and new intent enabled.

File: src/main/java/net/dv8tion/jda/api/requests/GatewayIntent.java [hunk 2]
```diff
@@ -407,7 +418,10 @@ else if (GenericUserUpdateEvent.class.isAssignableFrom(event) || GenericGuildMem
             else if (GuildBanEvent.class.isAssignableFrom(event) || GuildUnbanEvent.class.isAssignableFrom(event) || GuildAuditLogEntryCreateEvent.class.isAssignableFrom(event))
                 intents.add(GUILD_MODERATION);
             else if (GenericEmojiEvent.class.isAssignableFrom(event) || GenericGuildStickerEvent.class.isAssignableFrom(event))
+            {
                 intents.add(GUILD_EMOJIS_AND_STICKERS);
+                intents.add(GUILD_EXPRESSIONS);
+            }
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/requests/GatewayIntent.java:424]:
```suggestion
                intents.add(GUILD_EXPRESSIONS);
```

**Pull request merged**